src/client/metadataApiDeploy.ts

```diff
@@ -370,6 +374,46 @@ const deleteNotFoundToFileResponses =
           : [];
       });
 
+const serverResponseNotFoundLocally =
+  (cs: ComponentSet) =>
+  (messageMap: Map<string, DeployMessage[]>): FileResponse[] => {
+    const sourceKeys = cs.getSourceComponents().toArray().map(toKey);
```

Comment:
for a large deploy, this'll be a bit more performant if you make sourceKeys a `set` and use `.has(k)`